paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -527,7 +537,14 @@ def wait_for_deployment(service, deploy_group, git_sha, soa_dir, timeout):
 
     total_instances = 0
     clusters_data = []
+    api_endpoints = load_system_paasta_config().get_api_endpoints()
```

Comment:
I'm a bit confused at why we have to load the api_endpoints thing so early here.

Is there a function deeper down that actually looks up the api_endpoints and would be a more natural place to raise this exception?